packages/synapse-react-client/src/synapse-queries/entity/useExportToTerra.test.tsx

```diff
@@ -0,0 +1,142 @@
+import { describe, it, expect, vi, beforeEach, Mock } from 'vitest'
+import { renderHook, act } from '@testing-library/react'
+import {
+  QueryBundleRequest,
+  SelectColumn,
+} from '@sage-bionetworks/synapse-types'
+import * as SynapseConstants from '@/utils/SynapseConstants'
+import { DEFAULT_PAGE_SIZE } from '@/utils/SynapseConstants'
+import { useExportToTerra } from './useExportToTerra'
+
+const mockCreatePfb = vi.fn()
+
+vi.mock('../table/useDownloadTable', () => ({
+  useDownloadTableQueryResultAsPFB: () => ({ mutateAsync: mockCreatePfb }),
+}))
+
+import { displayToast } from '../../components/ToastMessage/ToastMessage'
+vi.mock('../../components/ToastMessage/ToastMessage', () => ({
+  displayToast: vi.fn(),
+}))
+
+import { parseEntityIdFromSqlStatement } from '../../utils/functions/SqlFunctions'
+vi.mock('../../utils/functions/SqlFunctions', () => ({
+  parseEntityIdFromSqlStatement: vi.fn(),
+}))
+
+import { addDrsUriToSql } from './useExportToCavatica'
+vi.mock('./useExportToCavatica', () => ({
+  addDrsUriToSql: vi.fn(),
+}))
+
+const originalSql = 'SELECT * FROM syn123'
+const entityId = 'syn123'
+const transformedSql = 'SELECT id, name FROM syn123 /* +drs */'
+
+const testQueryRequest: QueryBundleRequest = {
+  concreteType: 'org.sagebionetworks.repo.model.table.QueryBundleRequest',
+  entityId: entityId,
+  partMask:
+    SynapseConstants.BUNDLE_MASK_QUERY_COLUMN_MODELS |
+    SynapseConstants.BUNDLE_MASK_QUERY_FACETS |
+    SynapseConstants.BUNDLE_MASK_QUERY_RESULTS |
+    SynapseConstants.BUNDLE_MASK_QUERY_COUNT,
+  query: {
+    sql: originalSql,
+    limit: DEFAULT_PAGE_SIZE,
+    offset: 0,
+  },
+}
+describe('useExportToTerra ()', () => {
+  beforeEach(() => {
+    vi.resetAllMocks()
+    ;(parseEntityIdFromSqlStatement as Mock).mockReturnValue(entityId)
+    ;(addDrsUriToSql as unknown as Mock).mockReturnValue(transformedSql)
```

Comment:
Removes unnecessary cast and provides a `mock` object that is aware of the original type

```suggestion
    mocked(parseEntityIdFromSqlStatement).mockReturnValue(entityId)
    mocked(addDrsUriToSql).mockReturnValue(transformedSql)
```